src/redux/actions/order.js

```diff
@@ -23,6 +28,51 @@ const add = () => {
   }
 }
 
+const txStatusCheckAndUpdateOrder = order => {
+  return async (dispatch, getState, { getFirestore }) => {
+    const firestore = getFirestore()
+
+    if (!order.txConfirmed && order.txHash) {
+      try {
+        const status = await web3Instance.isTxConfirmed(order.txHash)
+        await firestore
```

Comment:
Not sure how firebase handles overwrite with the same data, but just to be safe I'd add:
```js
if (!txConfirmed) {
  return
}
```